### ISSUE TITLE
Support nested rules

### ISSUE THREAD

**Issue by davidspiess** [mikeerickson/validatorjs#67]
It would be great if i could define nested rules for nested objects

``` js
const rules = {
    firstname: 'required',
    lastname: 'required',
    email: 'required|email',
    emailConfirm: 'required|same:email',
    payment: {
        selection: 'required'
    },
    company: {
        name: 'required',
        taxnumber: 'required'
    }
};
```

At the moment i had to write a helper function for that

``` js
const createValidator = (values, rules) => {
    const validator = new Validator(values, rules);
    validator.passes();
    return validator.errors.all();
};

const validate = values => {
    let errors = {};

    errors = createValidator(values, rules);
    errors.payment = createValidator(values.payment, rules.payment);
    errors.company = createValidator(values.company, rules.company);

    return errors;
};
```

**Comment by iamdtang** [comment]
Do you know if Laravel's Validator supports this?

**Comment by davidspiess** [comment]
It indeed does.

The same example as above in Laravel:

``` php
$this->validate($request, [
    'firstname' => 'required',
    'lastname' => 'required',
    'email' => 'required|email',
    'emailConfirm' => 'required|same:email',
    'payment.selection' => 'required',
    'company.name' => 'required',
    'company.taxnumber' => 'required'
]);
```

**Comment by iamdtang** [comment]
ok I will try and take a stab at this one.

**Comment by LKay** [comment]
Are there any updates on this? Not sure if you are working on that or should I use workaround or implement it on my own.

**Comment by iamdtang** [comment]
I haven't had time. Would love some help. :)

**Comment by LKay** [comment]
I added the support for nested rules in the PR #92. 

Some basic tests also included. Check if it's fine and if could be included in the next version.

I would also recommend to rewrite library to ES6 code (or TypeScript), which improve readability and it's not difficult and time consuming task itself.

**Comment by iamdtang** [comment]
I'm open to using TypeScript. What do you think @garygreen?

**Comment by garygreen** [comment]
I would prefer to go with plain ES6 rather than TypeScript as I personally haven't used TypeScript (haven't seen the need, though I'm sure it has some useful features), using it could further slow the development of validatorjs because it could be seen as another barrier to contribute to the library - which I think we should encourage.

**Comment by LKay** [comment]
@garygreen The advantage of using TypeScript would be to have type safe code which would help to eliminate some potential bugs at the compilation step without the need to write tests. TS is subset of Javascript so any valid JS code will still be valid TS code and if linter rules will not be too strict anyone could contribute with ES5/6 code. 

Just for the record, I don't mind it to be ES6.

**Comment by garygreen** [comment]
Probably just my ignorance of TypeScript on display there, I don't have any strong opinions either way -- it'll be a learning experience for me.

**Comment by LKay** [comment]
Anyway, this discussion should be moved to separate issue. I'm using this library in few project and will be glad to make some es6/typescript skeleton as reference for contribute over the weekend.

**Comment by iamdtang** [comment]
Going to move this to a separate issue. This issue is closed and released in 2.1.0
